app/frontend/src/common/store/auth.js

```diff
@@ -1,18 +1,26 @@
 export const SET_KEYCLOAK = 'SET_KEYCLOAK'
+export const SET_KEYCLOAK_READY = 'SET_KEYCLOAK_READY'
```

Comment:
You may not need this state anymore (see comments below re. authentication now resolving promise correctly).

Reply:
I'll probably scrap this branch if you've come up with another solution. We'll have to touch base on what is left to be done/ what problems we still need to solve.